src/App.js

```diff
@@ -15,10 +16,41 @@ function App() {
     });
   }
 
+  function editCard() {
+    setCardState({
+      ...cardState,
+      edited: !cardState.edited,
+      checked: (cardState.checked ? !cardState.checked : cardState.checked)
+    });
+  }
+
+  function captionChange(event) {
+    setCardState({
+      ...cardState,
+      caption: event.target.value,
+    });
+  }
+
+  function textChange(event) {
+    setCardState({
+      ...cardState,
+      text: event.target.value
+    });
+  }
+
+  function cancelEditing() {
+    setCardState({
+      caption: 'Caption', text: 'Text...',
+      edited: !cardState.edited,
+      checked: (cardState.checked ? !cardState.checked : cardState.checked)
+    });
+  }
+
   return (
     <div className="App">
       <Header></Header>
-      <Card className="block" caption={cardState.caption} text={cardState.text} checked={cardState.checked} checkboxChange={checkboxChange}></Card>
+      <Card className="block" caption={cardState.caption} text={cardState.text} checked={cardState.checked} checkboxChange={checkboxChange}
+            edit={editCard} edited={cardState.edited} capEdit={captionChange} textEdit={textChange} cancel={cancelEditing}></Card>
```

Comment:
1. Far more easier to maintain passed data if they are combined, so here it would be much better to pass card data object.
2. A good practice - to name passed event handler properties with 'On' preffix (onDelete, onChange, onCreate, etc), so when working with downstream(children) component you could be aware which of props is an event handler to subscribe onto.